File: 2024/.tpl/main.ts
import { run } from '../shared/runtime.ts';

export type ParseOutput = ReturnType<typeof parse>;

export function parse(input: string) {
  const lines = input.trim().split('\n');

  return lines;
}

export function solvePart1(input: ParseOutput) {

}

export function solvePart2(input: ParseOutput) {

}

if (import.meta.main) {
  run(parse, solvePart1, solvePart2);
}

File: 2024/.tpl/main.go
package main

import (
	"strings"

	"github.com/janvoracek/advent-of-code/aoc/2024/shared"
)

type ParseOutput = []string

func parse(input string) ParseOutput {
	return strings.Split(strings.TrimSpace(input), "\n")
}

func solvePart1(input ParseOutput) any {
	return 0
}

func solvePart2(input ParseOutput) any {
	return 0
}

func main() {
	shared.Run(parse, solvePart1, solvePart2)
}
